lib/filesystem/Filesystem.cpp

```diff
@@ -117,7 +117,7 @@ void CFilesystemGenerator::loadJsonMap(const std::string &mountPoint, const Json
 	if (filename)
 	{
 		auto configData = CResourceHandler::get("initial")->load(ResourceID(URI, EResType::TEXT))->readAll();
-		const JsonNode configInitial((char*)configData.first.get(), configData.second);
+		const JsonNode configInitial(reinterpret_cast<char *>(configData.first.get()), configData.second);
```

Comment:
What is the difference between reinterpret_cast and static_cast?

Reply:
static_cast generally converts between convertible types (e.g int ->float, or classes within same type hierarchy)
reinterpret_cast allows well, reinterpreting variable as another, unrelated type.

So, for example static_cast<int>(float) will convert float to integer, using standard rules (drop fractional part).
static_cast'ing int* to float* would fail since these types are not related
Meanwhile, reinterpret_cast<int*>(float*) will give you your pointer to int, however data under that pointer will not be modified and still contain floating point representation in its IEEE form.

Reply:
Here we required to use reinterpret_cast, because compiler does not allows us to use static_cast,